.github/workflows/trivy-nightly-scan.yaml

```diff
@@ -1,37 +1,51 @@
-name: Trivy Nightly Scan
+name: Trivy Nightly Build Scan
 on:
   schedule:
     - cron: '0 2 * * *' # run at 2 AM UTC
-
 permissions: read-all 
 jobs:
   nightly-scan:
-    name: Trivy Scan nightly
+    name: Trivy Build Scan nightly
     strategy:
       fail-fast: false
       matrix:
         # maintain the versions of etcd that need to be actively
         # security scanned
-        versions: [v3.5.6]
+        target:
+          - amd64
+          - arm64
+          - ppc64le
+          - s390x
     permissions:
       security-events: write  # for github/codeql-action/upload-sarif to upload SARIF results
-
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
-        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
         with:
           ref: release-3.5
-      - name: Run Trivy vulnerability scanner
-        uses: aquasecurity/trivy-action@9ab158e8597f3b310480b9a69402b419bc03dbd5 # master
+      - uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
         with:
-          image-ref: 'gcr.io/etcd-development/etcd:${{ matrix.versions }}'
+          go-version: "1.19.6"
+      - run: |
+            ./build.sh
```

Comment:
We can remove this entire step/ There is no need to build the binaries or docker images, we can re-use the one created as part of release-workflow (once you have ported this to etcd:main). See this [code](https://github.com/etcd-io/etcd/blob/8b93e0e46bb0a698e425d8a4b0ba82f7510fd0c8/scripts/test_images.sh#L39-L56) for reference.